test/api/config.go

```diff
@@ -0,0 +1,143 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"bufio"
+	"os"
+	"strings"
+	"time"
+)
+
+type TestConfig struct {
+	BaseURL            string
+	IdentityBaseURL    string
+	AuthToken          string
+	RequestTimeout     time.Duration
+	TestTimeout        time.Duration
+	OrgID              string
+	ProjectID          string
+	SecondaryProjectID string
+	RegionID           string
+	SecondaryRegionID  string
+	FlavorID           string
+	ImageID            string
+	SkipIntegration    bool
+	DebugLogging       bool
+	LogRequests        bool
+	LogResponses       bool
+}
+
+// all of the errors that show below should only show locally, not in the CI/CD pipeline since we will add secrets.
+func LoadTestConfig() *TestConfig {
+	config := &TestConfig{
+		BaseURL:            "REQUIRED: Set API_BASE_URL in the .env file",
+		IdentityBaseURL:    "REQUIRED: Set IDENTITY_BASE_URL in the .env file",
+		RequestTimeout:     30 * time.Second,
+		TestTimeout:        20 * time.Minute, // 20 minutes is the default timeout for the tests for now, as thats what I had it in SOS
+		OrgID:              "REQUIRED: Set TEST_ORG_ID in the .env file",
+		ProjectID:          "REQUIRED: Set TEST_PROJECT_ID in the .env file",
+		SecondaryProjectID: "REQUIRED: Set TEST_SECONDARY_PROJECT_ID in the .env file",
+		RegionID:           "REQUIRED: Set TEST_REGION_ID in the .env file",
+		SecondaryRegionID:  "REQUIRED: Set TEST_SECONDARY_REGION_ID in the .env file",
+		FlavorID:           "REQUIRED: Set TEST_FLAVOR_ID in the .env file",
+		ImageID:            "REQUIRED: Set TEST_IMAGE_ID in the .env file",
+	}
+
+	envVars := loadEnvFile()
+	loadConfigFromEnv(config, envVars)
+
+	return config
+}
+
+// loadConfigFromEnv loads configuration values from environment variables.
+func loadConfigFromEnv(config *TestConfig, envVars map[string]string) {
+	// String values
+	setStringValue(&config.BaseURL, getEnvValue(envVars, "API_BASE_URL"))
+	setStringValue(&config.IdentityBaseURL, getEnvValue(envVars, "IDENTITY_BASE_URL"))
+	setStringValue(&config.AuthToken, getEnvValue(envVars, "API_AUTH_TOKEN"))
+	setStringValue(&config.OrgID, getEnvValue(envVars, "TEST_ORG_ID"))
+	setStringValue(&config.ProjectID, getEnvValue(envVars, "TEST_PROJECT_ID"))
+	setStringValue(&config.SecondaryProjectID, getEnvValue(envVars, "TEST_SECONDARY_PROJECT_ID"))
+	setStringValue(&config.RegionID, getEnvValue(envVars, "TEST_REGION_ID"))
+	setStringValue(&config.SecondaryRegionID, getEnvValue(envVars, "TEST_SECONDARY_REGION_ID"))
+	setStringValue(&config.FlavorID, getEnvValue(envVars, "TEST_FLAVOR_ID"))
+	setStringValue(&config.ImageID, getEnvValue(envVars, "TEST_IMAGE_ID"))
+}
+
+// setStringValue sets a string value if the env value is not empty.
+func setStringValue(target *string, value string) {
+	if value != "" {
+		*target = value
+	}
+}
+
+func loadEnvFile() map[string]string {
```

Comment:
Ugh... looks like reinvention of the wheel!  Can we not make use of https://pkg.go.dev/gopkg.in/ini.v1 or something like https://github.com/spf13/viper??  Also I don't see any error handling for when a required field isn't specified.  May just need to read further...